test/vagrant-service-manager/proxy_test.rb

```diff
@@ -0,0 +1,109 @@
+require_relative '../test_helper'
+require 'vagrant/util/downloader'
+
+module VagrantPlugins
+  module ServiceManager
+    describe 'Proxy Test' do
+      before do
+        @machine = fake_machine
+        @ui = FakeUI.new
+        # set test path
+        @plugin_test_path = "#{@machine.env.data_dir}/service-manager/test"
+        ServiceManager.temp_dir = "#{@plugin_test_path}/temp"
+        ServiceManager.bin_dir = "#{@plugin_test_path}/bin"
+
+        @configs = { htpasswd_file: "#{ServiceManager.temp_dir}/.htpasswd" }
+      end
+
+      after do
+        FileUtils.rmtree(@plugin_test_path) if File.directory? @plugin_test_path
+      end
+
+      describe 'Unauthenticated proxy environment' do
+        before do
+          @string_stream = StringIO.new
+          @options = { type: :docker, '--cli-version' => '1.10.0' }
+          @handler = ADBDockerBinaryHandler.new(@machine, @machine.env, @options)
+        end
+
+        after do
+          Thread.kill @download_thread
+          Thread.kill @proxy_thread
+          @proxy_server.shutdown
+        end
+
+        it 'should send client binary download request' do
+          @handler.build_download_url
+          @handler.validate_url
+          @handler.build_archive_path
+          @handler.ensure_binary_and_temp_directories
+
+          @proxy_thread = Thread.new do
```

Comment:
So why do you do this in the test and not in setup. This blows up the testing code and makes it hard to see what the actual test is about.